File: src/main/kotlin/com/back/domain/files/files/service/FileStorageService.java
package com.back.domain.files.files.service;

import org.springframework.core.io.Resource;

public interface FileStorageService {
    String storeFile(byte[] fileContent, String originalFilename, String contentType, String subFolder);
    void deletePhysicalFile(String fileUrl);
    Resource loadFileAsResource(String fileUrl);
}

File: src/main/kotlin/com/back/domain/files/files/service/FileStorageService.kt
package com.back.domain.files.files.service

import org.springframework.core.io.Resource

interface FileStorageService {
    fun storeFile(fileContent: ByteArray, originalFilename: String, contentType: String, subFolder: String): String
    fun deletePhysicalFile(fileUrl: String)
    fun loadFileAsResource(fileUrl: String): Resource
}
